internal/auth/jwtauth.go

```diff
@@ -139,6 +140,20 @@ func WithPermissionsContext(ctx context.Context, claims UserPermissions) context
 	return context.WithValue(ctx, tokenContextKey, claims)
 }
 
+// GetUserSubjectFromContext returns the user subject from the context, or nil
+func GetUserSubjectFromContext(ctx context.Context) *string {
+	subject, ok := ctx.Value(userSubjectContextKey).(string)
```

Comment:
If we enforce that `subject` is always non-empty, we could just return an empty string, which might simplify other code.